src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/events/PluginPatchsetCreatedEvent.java

```diff
@@ -203,6 +221,10 @@ public boolean shouldTriggerOn(GerritTriggeredEvent event) {
         if (excludeWipState && ((PatchsetCreated)event).getChange().isWip()) {
             return false;
         }
+        if (null != this.commitMessageContainsRegEx && !this.commitMessageContainsRegEx.isEmpty()) {
+            Pattern pattern = Pattern.compile(this.commitMessageContainsRegEx, Pattern.DOTALL | Pattern.MULTILINE);
```

Comment:
This method can potentially be called many times per second. Better to cache it in a `transient` field and only compile a new instance if the field is `null`.

Reply:
I will update